rest/rest-server/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/RMRestInterface.java

```diff
@@ -167,6 +168,18 @@ NSState defineNodeSource(@HeaderParam("sessionid") String sessionId, //NOSONAR
             @FormParam("policyFileParameters") String[] policyFileParameters,
             @FormParam("nodesRecoverable") String nodesRecoverable) throws NotConnectedException;
 
+    @PUT
+    @Path("nodesource/edit")
+    @Produces("application/json")
+    NSState editNodeSource(@HeaderParam("sessionid") String sessionId,
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Method has 9 parameters, which is greater than 7 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00107)